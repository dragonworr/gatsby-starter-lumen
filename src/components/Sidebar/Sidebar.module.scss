@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.sidebar {
  width: 100%;
  &__inner {
    padding: 25px 20px 0;
    position: relative;
  }
}

@include breakpoint-sm {
  .sidebar {
    lost-column: 5/12;
    &__inner {
      padding: 30px 20px 0;
      &:after {
        background: $color-gray-border;
        background: linear-gradient(
          to bottom,
          $color-gray-border 0%,
          $color-gray-border 48%,
          $color-white 100%
        );
        bottom: 0;
        content: "";
        height: 540px;
        position: absolute;
        right: -10px;
        top: 30px;
        width: 1px;
      }
    }
  }
}

@include breakpoint-md {
  .sidebar {
    lost-column: 1/3;
    &__inner {
      padding: 40px;
    }
  }
}
