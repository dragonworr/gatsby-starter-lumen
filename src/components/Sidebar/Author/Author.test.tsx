import React from "react";
import renderer from "react-test-renderer";

import { Author } from "@/components/Sidebar/Author";
import * as mocks from "@/mocks";

describe("Author", () => {
  const props = { isIndex: false, author: mocks.author };

  it("renders correctly", () => {
    const tree = renderer.create(<Author {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
