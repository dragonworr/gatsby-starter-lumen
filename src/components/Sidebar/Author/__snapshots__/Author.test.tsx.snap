// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
<div
  className="author"
>
  <a
    href="/"
  >
    <img
      alt="John Doe"
      className="author__photo"
      height="75"
      src="/static/photo.jpg"
      width="75"
    />
  </a>
  <h2
    className="author__title"
  >
    <a
      className="author__title-link"
      href="/"
    >
      John Doe
    </a>
  </h2>
  <p
    className="author__subtitle"
  >
    Pellentesque odio nisi, euismod in, pharetra a, ultricies in, diam. Sed arcu.
  </p>
</div>
`;
